File: SDK/QDSJavaSDK/src/qds/api/QDSAPI.java
package qds.api;

import java.lang.String;

import com.sun.jna.Library;
import com.sun.jna.Native;

//基础接口
public class QDSAPI {
	

	public interface CLibrary extends Library
    {
		void CreateInstance();
		int RegisterService(String ip, char port);
		int Login(String publicKey, String secretKey, boolean bWAN);
		int Login2(String userName, boolean bWAN);
		int Subscribe(int msgType, String codeList);
		int Unsubscribe(int msgType, String codeList);
		int UnsubscribeAll();
		void ReleaseInstance();
    }
	
	private static CLibrary lib = (CLibrary)Native.loadLibrary("wQDSApi.dll",CLibrary.class); 
	
	public static void CreateInstance()
	{
		lib.CreateInstance();
	}
	
	public static int RegisterService(String ip, char port)
	{
		return lib.RegisterService(ip, port);
	}
	
	public static int Login(String publicKey, String secretKey, boolean bWAN)
	{
		return lib.Login(publicKey, secretKey, bWAN);
	}
	
	public static int Login2(String userName, boolean bWAN)
	{
		return lib.Login2(userName, bWAN);
	}
	
	public static int Subscribe(int msgType, String codeList)
	{
		return lib.Subscribe(msgType, codeList);
	}
	
	public static int Unsubscribe(int msgType, String codeList)
	{
		return lib.Unsubscribe(msgType, codeList);
	}
	
	public static int UnsubscribeAll()
	{
		return lib.UnsubscribeAll();
	}
	
	public static void ReleaseInstance()
	{
		lib.ReleaseInstance();
	}
	
}

File: SDK/QDSPythonSDK/QDSPythonSDK/QDSAPI.py
from QDSApiCallbackBase import QDSApiCallbackBase
from QDSApiCallbackBaseDelegate import cLib

class QDSAPI:
    @classmethod
    def CreateInstance(Obj):
        return cLib.CreateInstance()
    
    @classmethod
    def RegisterService(Obj, ip, port):
        return cLib.RegisterService(ip, port)

    @classmethod
    def Login(Obj, publicKey, secretKey, bWAN):
        return cLib.Login(publicKey, secretKey, bWAN)

    @classmethod
    def Login2(Obj, userName, bWAN):
        return cLib.Login2(userName, bWAN)

    @classmethod
    def Subscribe(Obj, msgType, codeList):
        return cLib.Subscribe(msgType, codeList)

    @classmethod
    def Unsubscribe(Obj, msgType, codeList):
        return cLib.Unsubscribe(msgType, codeList)

    @classmethod
    def UnsubscribeAll(Obj):
        return cLib.UnsubscribeAll()

    @classmethod
    def ReleaseInstance(Obj):
        return cLib.ReleaseInstance()